src/deltacode/utils.py

```diff
@@ -145,27 +153,29 @@ def update_modified_from_copyright_info(delta):
     one or more categories to its 'factors' attribute if there has
     been a copyright change.
     """
-    new_copyrights = delta.new_file.copyrights or []
-    old_copyrights = delta.old_file.copyrights or []
-
-    if delta.new_file.has_copyrights() and not delta.old_file.has_copyrights():
-        delta.update(10, 'copyright info added')
-        return
-    if not delta.new_file.has_copyrights() and delta.old_file.has_copyrights():
-        delta.update(10, 'copyright info removed')
-        return
-
-    new_holders = set(holder for copyright in new_copyrights for holder in copyright.holders)
-    old_holders = set(holder for copyright in old_copyrights for holder in copyright.holders)
 
-    if new_holders != old_holders:
-        delta.update(5, 'copyright change')
+    try:
+        new_copyrights = delta.new_file.copyrights or []
+        old_copyrights = delta.old_file.copyrights or []
+        if delta.new_file.copyrights and not delta.old_file.copyrights:
+            delta.update(10, 'copyright info added')
+            return
+        if not delta.new_file.copyrights and delta.old_file.copyrights:
+            delta.update(10, 'copyright info removed')
+            return
+        new_holders = set(holder for copyright in new_copyrights for holder in copyright['holders'])
+        old_holders = set(holder for copyright in old_copyrights for holder in copyright['holders'])
+        if new_holders != old_holders:
+            delta.update(5, 'copyright change')
+    except Exception as exception:
```

Comment:
What line of code in the try block raises the exceptions? and is it a single exception or more than one.

Reply:
No @majurg it was an exception of more than one :(

Reply:
Ok we need to refactor this a bit, I do not like catching any exception and doing absolutely nothing. What exact exceptions are being raised here?

Reply:
The error is raised when the `Resource` object is not having the `copyright` attributes

Reply:
Ok, we could put a guard in before performing operations on the `.copyright` data to avoid having this exception raised

Reply:
Yes that could be done doing it shortly

Reply:
Changed in 85c85edf2aa669531021fba34d9c186810022558